plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/MockStorage.java

```diff
@@ -256,6 +266,11 @@ public void setChunkSize(int chunkSize) {
 
                 @Override
                 public int write(ByteBuffer src) throws IOException {
+                    // Only fail a blob once on a 410 error since the error is so unlikely in practice
+                    if (simulated410s.add(blobInfo) && random.nextBoolean()) {
```

Comment:
can we test this with @tlrx's recent HTTP mock of GCS?

Reply:
That's the goal, but for now repository integration tests do not support GCS resumable uploads (as we don't generate large enough segment files). 

I can add GCS-like retries test at the blob store level like I did for S3 (see S3BlobContainerRetriesTests) and test this retry logic with it.

Reply:
Left this for @tlrx now if that's ok :) No need for two people to mess with the API mock code concurrently I guess.

Reply:
It took some time but I finally opened https://github.com/elastic/elasticsearch/pull/46968, which adds some random `410-Gone` errors when testing resume upload (see `testWriteLargeBlob` test)